smartsim/_core/_install/buildenv.py

```diff
@@ -61,6 +72,29 @@ class SetupError(Exception):
     See setup.py for more
     """
 
+class VersionConflictError(SetupError):
+    """An error for when version numbers of some library/package/program/etc
+    do not match and build may not be able to continue
+    """
+
+    def __init__(
+        self,
+        name: str,
+        current_version: "Version_",
+        target_version: "Version_",
+        msg: t.Optional[str] = None,
+    ) -> None:
+        if msg is None:
```

Comment:
Do we want to override `__str__` and `__repr__` on SetupError children instead of having msg built in a constructor?

Reply:
I'd argue that we should still pass he message through the `super` constructor just so the the `Exception.message` attr gets set correctly. Iirc the default `__str__` basically amounts to
```py
def __str__(self):
    return str(self.message)
```
so I'm not sure that I would see the benefit of the override.

That said, I could be convinced that the construction of the message string that gets passed to `super().__init__` does clutter up the constructor. Maybe we should just move it into its own static helper method?